typescript/integrations.d.ts

```diff
@@ -22,90 +22,106 @@ type IntegrationBase = {
   createdAt: Date
 }
 
-export type IntegrationTrimEnvironment = {
+export type IntegrationSecret = {
+  id: number
+  createdAt: string
+  updatedAt: string
+  cipherText: string
+  encryptionKeyId: string
+}
+
+export type SavedSecret = {
+  secretPointer: number
+}
+export type NewSecret = string
+
+export type Secret<T> = T extends SavedSecret | NewSecret ? T : SavedSecret
+
+export type IntegrationTrimEnvironment<S = SavedSecret> = {
   id: string
   label: string
   baseUrl: string
   username: string
-  password: string
+  password: Secret<S>
```

Comment:
I would prefer to make a breaking change to all of these types and remove the "secret" property (`password` in this case) and add a new property called `passwordPointer: number`. This this way we dont need to do type checking on the `password` property to determine if they want to reset it or not, we just make either `password` or `passwordPointer` required as part of the request to upsert the integration.

Reply:
I would prefer not to do that. This way is nicer for validation on both the server and client IMO.

Reply:
It's also less types we need to define as `EnvironmentX` and `NewEnvironmentX`